ui/app/components/region-switcher.js

```diff
@@ -13,16 +13,22 @@ export default class RegionSwitcher extends Component {
   @service system;
   @service router;
   @service store;
+  @service token;
 
   @computed('system.regions')
   get sortedRegions() {
     return this.get('system.regions').toArray().sort();
   }
 
-  gotoRegion(region) {
-    this.router.transitionTo('jobs', {
-      queryParams: { region },
-    });
+  async gotoRegion(region) {
+    // Reset token
+    // this.token.reset();
+    // Note: redundant but as long as we're using PowerSelect, the implicit set('activeRegion')
+    // is not something we can await, so we do it explicitly here.
+    this.system.set('activeRegion', region);
+    await this.get('token.fetchSelfTokenAndPolicies').perform().catch();
+
+    this.router.transitionTo({ queryParams: { region } });
```

Comment:
Note: made the decision not to send you back to the jobs index when you switch regions anymore. This was a ~6 year old decision that made sense at the time, but we now have many more routes that may be context-shared between regions.

For example, if I sit on the /administration/policies page and switch between regions, I should now see my policies list simply update. Before, I'd have been jostled, have to click Administration, then have to click Policies again.

The drawback from this is that sometimes routes (like a particular task in a particular job) are NOT shared between regions. Switching a region on a page like this will put you on a (friendly) 404 page. This is a tradeoff that I think is worthwhile.

Reply:
That's reasonable, for sure. Another thing to consider is that all the administrative routes are cross-region (because the authoritative region owns all these objects), whereas everything else is single region. But having two different behaviors depending on context might be more confusing than it's worth.